src/main/java/frc/robot/subsystems/BallPath/Shooter/ShooterImpl.java

```diff
@@ -2,17 +2,83 @@
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 public class ShooterImpl extends RepeatingPooledSubsystem implements Shooter {
-    public ShooterImpl() {
+    
+    private TalonSRX turretMotor;
+    private TalonFX shooterMotor;
+    private TalonSRX hoodMotor;
+
+    private double kp = 0.00035;
+    private double ki = 0.000075;
+    // private final double kd = -0.0002;  Value that was being tested as of February 3, 2022.
+    private double kd = 0.00003;
+
+    private double setPointHood;
+    private double setPointShooter;
+    private double setPointRotation;
+    private double setPointShooterPID;
+
+    private double shooterEncoderReadingPosition;
+    double shooterEncoderReadingVelocity;
+    private double turretHoodPosition;
+    private double turretEncoderReadingPosition;
+    private double turretHoodVelocity;
+    private double turretEncoderReadingVelocity;
+
+    private boolean turretReady = false;
+    private boolean hoodReady = false;
+
+    double currentOutput;
+
+    private final PIDController shooterPid = new PIDController(kp, ki, kd);
+
+    public ShooterImpl(TalonSRX turretMotor, TalonFX shooterMotor, TalonSRX hoodMotor) {
         super(10, TimeUnit.MILLISECONDS);
         // TODO Auto-generated constructor stub
+        this.turretMotor = turretMotor;
+        this.shooterMotor = shooterMotor;
+        this.hoodMotor = hoodMotor;
+
+        shooterEncoderReadingPosition = shooterMotor.getSelectedSensorPosition();
+        shooterEncoderReadingVelocity = shooterMotor.getSelectedSensorVelocity();
+        turretHoodPosition = hoodMotor.getSelectedSensorPosition();
+        turretHoodVelocity = hoodMotor.getSelectedSensorVelocity();
+        turretEncoderReadingPosition = this.turretMotor.getSelectedSensorPosition();
+        turretEncoderReadingVelocity = this.turretMotor.getSelectedSensorVelocity();
+
+        SmartDashboard.putNumber("Shooter Encoder reading position", shooterEncoderReadingPosition);
+        SmartDashboard.putNumber("Shooter Encoder Reading Velocity", shooterEncoderReadingVelocity);
+        SmartDashboard.putNumber("Turret Encoder Reading Position", turretEncoderReadingPosition);
+        SmartDashboard.putNumber("Turret Encoder Reading Velocity", turretEncoderReadingVelocity);
+        SmartDashboard.putNumber("Turret Hood Encoder reading Position", turretHoodPosition);
+        SmartDashboard.putNumber("Turret Hood Encoder Reading Velocity", turretHoodVelocity);
+
     }
 
     @Override
-    public void defineResources(){}
+    public void defineResources(){
+        require(turretMotor);
+        require(shooterMotor);
+        require(hoodMotor);
+
+        require(shooterEncoderReadingPosition);
+        require(shooterEncoderReadingVelocity);
+        require(turretHoodPosition);
+        require(turretHoodVelocity);
+        require(turretEncoderReadingPosition);
+        require(turretEncoderReadingVelocity);
```

Comment:
Fields like these positions and velocities don't need to be required, only things like motor controllers or other subsytems which are passed into this subsystem's constructor and therefore might be shared instances.